3party/CMakeLists.txt

```diff
@@ -2,7 +2,10 @@
 # Compatibility with CMake < 3.5 will be removed from a future version of CMake.
 set(CMAKE_WARN_DEPRECATED OFF CACHE BOOL "" FORCE)
 
-if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
+if (WITH_SYSTEM_PROVIDED_3PARTY)
```

Comment:
Do we track the list of system-provided 3p libs that should be installed when WITH_SYSTEM_PROVIDED_3PARTY is enabled? If no, then let's create it and document it somewhere.

Reply:
A semi automatic way to find that out:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index aaf7072413..5379636fae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -337,6 +337,9 @@ add_subdirectory(tracking)
 add_subdirectory(traffic)
 add_subdirectory(transit)
 
+get_property(importTargets DIRECTORY "${CMAKE_SOURCE_DIR}" PROPERTY IMPORTED_TARGETS)
+message("Imported Targets: ${importTargets}")
+
 if (PLATFORM_DESKTOP)
   if (BUILD_MAPSHOT)
     add_subdirectory(mapshot)
```

gives (with additional line breaks and sorted):
```
Imported Targets:
Freetype::Freetype;
ICU::data;
ICU::i18n;
ICU::uc;
OpenGL::EGL;
OpenGL::GL;
OpenGL::GLES2;
OpenGL::GLES3;
OpenGL::GLU;
OpenGL::GLX;
OpenGL::OpenGL;
Python3::Interpreter
Qt6::Core;
Qt6::CorePrivate;
Qt6::DBus;
Qt6::DBusPrivate;
Qt6::GlobalConfig;
Qt6::GlobalConfigPrivate;
Qt6::Gui;
Qt6::GuiPrivate;
Qt6::Network;
Qt6::NetworkPrivate;
Qt6::OpenGL;
Qt6::OpenGLPrivate;
Qt6::OpenGLWidgets;
Qt6::OpenGLWidgetsPrivate;
Qt6::Platform;
Qt6::PlatformAppInternal;
Qt6::PlatformCommonInternal;
Qt6::PlatformModuleInternal;
Qt6::PlatformPluginInternal;
Qt6::PlatformToolInternal;
Qt6::QComposePlatformInputContextPlugin;
Qt6::QGifPlugin;
Qt6::QGlibNetworkInformationPlugin;
Qt6::QICNSPlugin;
Qt6::QICOPlugin;
Qt6::QIbusPlatformInputContextPlugin;
Qt6::QJpegPlugin;
Qt6::QLibInputPlugin;
Qt6::QLinuxFbIntegrationPlugin;
Qt6::QMinimalIntegrationPlugin;
Qt6::QMngPlugin;
Qt6::QNetworkManagerNetworkInformationPlugin;
Qt6::QOffscreenIntegrationPlugin;
Qt6::QSvgIconPlugin;
Qt6::QSvgPlugin;
Qt6::QTgaPlugin;
Qt6::QTiffPlugin;
Qt6::QTlsBackendCertOnlyPlugin;
Qt6::QTlsBackendOpenSSLPlugin;
Qt6::QTuioTouchPlugin;
Qt6::QVkKhrDisplayIntegrationPlugin;
Qt6::QVncIntegrationPlugin;
Qt6::QWaylandEglPlatformIntegrationPlugin;
Qt6::QWaylandIntegrationPlugin;
Qt6::QWbmpPlugin;
Qt6::QWebpPlugin;
Qt6::QXcbEglIntegrationPlugin;
Qt6::QXcbGlxIntegrationPlugin;
Qt6::QXcbIntegrationPlugin;
Qt6::QXdgDesktopPortalThemePlugin;
Qt6::Svg;
Qt6::SvgPrivate;
Qt6::Widgets;
Qt6::WidgetsPrivate;
Qt6::Xml;
Qt6::XmlPrivate;
Qt6::cmake_automoc_parser;
Qt6::moc;
Qt6::qdbuscpp2xml;
Qt6::qdbusxml2cpp;
Qt6::qlalr;
Qt6::qmake;
Qt6::qtpaths;
Qt6::qvkgen;
Qt6::rcc;
Qt6::syncqt;
Qt6::tracegen;
Qt6::tracepointgen;
Qt6::uic;
Qt::Core;
Qt::CorePrivate;
Qt::DBus;
Qt::DBusPrivate;
Qt::GlobalConfig;
Qt::GlobalConfigPrivate;
Qt::Gui;
Qt::GuiPrivate;
Qt::Network;
Qt::NetworkPrivate;
Qt::OpenGL;
Qt::OpenGLPrivate;
Qt::OpenGLWidgets;
Qt::OpenGLWidgetsPrivate;
Qt::Platform;
Qt::PlatformAppInternal;
Qt::PlatformCommonInternal;
Qt::PlatformModuleInternal;
Qt::PlatformPluginInternal;
Qt::PlatformToolInternal;
Qt::Svg;
Qt::SvgPrivate;
Qt::Widgets;
Qt::WidgetsPrivate;
Qt::Xml;
Qt::XmlPrivate;
Qt::cmake_automoc_parser;
Qt::moc;
Qt::qdbuscpp2xml;
Qt::qdbusxml2cpp;
Qt::qlalr;
Qt::qmake;
Qt::qtpaths;
Qt::qvkgen;
Qt::rcc;
Qt::syncqt;
Qt::tracegen;
Qt::tracepointgen;
Qt::uic;
Threads::Threads;
Vulkan::Headers;
Vulkan::Vulkan;
Vulkan::glslangValidator;
Vulkan::glslc;
WrapAtomic::WrapAtomic;
WrapOpenGL::WrapOpenGL;
WrapVulkanHeaders::WrapVulkanHeaders;
XKB::XKB;
```

In which format do you prefer the dependencies? I would use a format that is used by any distribution, since a distribution package has to specify the dependencies anyways.

Reply:
A list of libraries to install with apt or other package managers may be helpful for creating packages.